index.js

```diff
@@ -89,6 +91,120 @@ async function create ({
   }
 }
 
+/**
+ * checkPeersSync.
+ *
+ * @description check if there is at least one peer fully synchronized (up to date) with our drive
+ * @async
+ * @param {Object} contentFeed hypercore content feed
+ * @param {Array} stats array containing { file: String, start: Number, end: Number}. start === stat.offset and end === blocks
+ * @param {[Number]} wait a number in miliseconds to use as a function timeout. Defaults to 120000 (2 minutes). OPTIONAL
+ */
+function checkPeersSync (contentFeed, stats, wait = 120000) {
+  if (!contentFeed) {
+    throw new Error('content feed is required')
+  }
+
+  const result = {}
+  result.promise = new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      result.reject(new Error(`timeout exceded: ${wait}ms`))
+    }, wait)
+
+    result.resolve = () => {
+      contentFeed.off('upload', checkSync)
+      clearTimeout(timer)
+      resolve()
+    }
+
+    result.reject = err => {
+      contentFeed.off('upload', checkSync)
+      clearTimeout(timer)
+      reject(err)
+    }
+  })
+
+  const checkSync = debounce(() => {
+    console.log({ peers: contentFeed.peers })
+    console.log({ stats })
+
+    for (const peer of contentFeed.peers) {
+      // Ignore peers without remote bitfields
+      if (!peer.remoteBitfield) continue
+
+      // iterate over ranges
+      let haveAll = true
+
+      for (const stat of stats) {
+        for (let idx = stat.start; idx <= stat.end; idx++) {
+          if (idx === 0) continue
+          if (!peer.remoteBitfield.get(idx)) {
+            haveAll = false
+            break
```

Comment:
Maybe we should return from the function instead of doing a `break`? If any blocks from the stats are `false` then we don't need to check further.